rasa/nlu/featurizers/dense_featurizer/mitie_featurizer.py

```diff
@@ -1,89 +1,126 @@
     DENSE_FEATURIZABLE_ATTRIBUTES,
     FEATURIZER_CLASS_ALIAS,
     TOKENS_NAMES,
 )
+from rasa.nlu.utils.mitie_utils import MitieModel
 from rasa.shared.nlu.constants import FEATURE_TYPE_SENTENCE, FEATURE_TYPE_SEQUENCE
 from rasa.utils.tensorflow.constants import MEAN_POOLING, POOLING
+from rasa.nlu.featurizers.dense_featurizer._mitie_featurizer import MitieFeaturizer
 
 if typing.TYPE_CHECKING:
     import mitie
 
+logger = logging.getLogger(__name__)
 
-class MitieFeaturizer(DenseFeaturizer):
-    @classmethod
-    def required_components(cls) -> List[Type[Component]]:
-        return [MitieNLP, Tokenizer]
+# TODO: This is a workaround around until we have all components migrated to
+# `GraphComponent`.
+MitieFeaturizer = MitieFeaturizer
 
-    defaults = {
-        # Specify what pooling operation should be used to calculate the vector of
-        # the complete utterance. Available options: 'mean' and 'max'
-        POOLING: MEAN_POOLING
-    }
 
-    def __init__(self, component_config: Optional[Dict[Text, Any]] = None) -> None:
-        super().__init__(component_config)
+class MitieFeaturizerGraphComponent(DenseFeaturizer2, GraphComponent):
+    """A class that featurizes using Mitie."""
 
-        self.pooling_operation = self.component_config["pooling"]
+    @staticmethod
+    def get_default_config() -> Dict[Text, Any]:
+        """Returns the component's default config."""
+        return {
+            **DenseFeaturizer2.get_default_config(),
+            "mitie_model": None,
+            # Specify what pooling operation should be used to calculate the vector of
+            # the complete utterance. Available options: 'mean' and 'max'
+            POOLING: MEAN_POOLING,
+        }
 
-    @classmethod
-    def required_packages(cls) -> List[Text]:
+    @staticmethod
+    def required_packages() -> List[Text]:
+        """Any extra python dependencies required for this component to run."""
         return ["mitie", "numpy"]
 
+    def __init__(
+        self,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> None:
+        """Instantiates a new `MitieFeaturizerGraphComponent` instance."""
+        super().__init__(execution_context.node_name, config)
+        self.config = config
+        self.pooling_operation = self.config["pooling"]
+
+    @classmethod
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> "MitieFeaturizerGraphComponent":
+        """Creates a new untrained component (see parent class for full docstring)."""
+        return cls(config, model_storage, resource, execution_context)
+
+    @classmethod
+    def validate_config(cls, config: Dict[Text, Any]) -> None:
+        """Validates that the component is configured properly."""
+        mitie_model = config["mitie_model"]
```

Comment:
This shouldn't come though the config but rather as a parameter for `process` and `process_training_data`.

Reply:
But `process` is definited in `Featurizer2` class, and it's signature is `def process(self, messages: List[Message]) -> List[Message]`

Reply:
True, but this is how the model will be provided to the MitieFeaturizer (like the other Mitie components). I guess we should remove it from `Featurizer2` as it won't match all the featurizers.

Reply:
Done!